### DISCUSSION TITLE
Base path support

### DISCUSSION THREAD

**Discussion by patjakubik** [withastro/starlight#1407] [Feature Requests]
### What version of `starlight` are you using?

0.16.0

### What is your idea?

I already have a marketing page to which I want to add Starlight-powered docs under /docs path. 

I worked around this by patching Starlight package with the following patch:

```diff
node_modules/@astrojs/starlight/components/SidebarSublist.astro
- href={entry.href}
+ href={"/docs" + entry.href}
```

```diff
node_modules/@astrojs/starlight/components/SiteTitle.astro
- const href = formatPath(Astro.props.locale || '/');
+ const href = formatPath(Astro.props.locale || '/docs');
```

```diff
node_modules/@astrojs/starlight/index.ts
- pattern: '[...slug]',
+ pattern: 'docs/[...slug]',
```

### Why is this feature necessary?

Without it it's hard to integrate Starlight with existing site

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Hi @patryk-smc — thanks for the post!

Just double checking you’ve seen some related links:
- [“Use Starlight at a subpath”](https://starlight.astro.build/manual-setup/#use-starlight-at-a-subpath) docs section
- https://github.com/withastro/starlight/discussions/966 discussion

In fact, I’d say this probably a duplicate of that discussion there, so if it’s OK with you, I’ll close this as a duplicate 🙌
